### PR TITLE
Issue 19

### PR DESCRIPTION
**Pull request by timpbrenner**
https://github.com/Hireology/erlen/issues/19

### CODE REVIEW

File: lib/erlen/schema/array_of.rb [hunk 1]
```diff
@@ -268,13 +260,23 @@ def normalize_element(element)
           if self.class.element_type <= Base && element.is_a?(Hash)
             self.class.element_type.new(element)
 
-          elsif self.class.element_type <= Base
+          elsif self.class.element_type <= Base && !(element.class <= Base)
```
Comment (davidan1981) [inline lib/erlen/schema/array_of.rb]: I'm not sure why this change matters.

File: lib/erlen/schema/array_of.rb [hunk 2]
```diff
@@ -268,13 +260,23 @@ def normalize_element(element)
...
             self.class.element_type.import(element)
 
           else
             element
           end
         end
+
+        def __validate_payload
```
Comment (davidan1981) [inline lib/erlen/schema/array_of.rb]: There might have to be `super` somewhere.

Comment (davidan1981) [inline lib/erlen/schema/array_of.rb]: Also I'm not sure why you moved from `validate` call to this overriding this method.

File: lib/erlen/schema/array_of.rb (hunk 1 above)
Comment (timpbrenner) [inline lib/erlen/schema/array_of.rb]: So what was happening is if I passed in a payload, it would create a new payload with the desired schema, and I think that's wrong. We can discuss.

File: lib/erlen/schema/array_of.rb (hunk 2 above)
Comment (timpbrenner) [inline lib/erlen/schema/array_of.rb]: This is so we can control the validate error
